src/test/directo.test.js

```diff
@@ -1,5 +1,84 @@
-const { retTrue } = require('../');
+const { Format, CommandResult } = require('../enums');
+const { Directo } = require('../directo');
 
-test('some placeholder test', () => {
-  expect(retTrue()).toBe(true);
+function badCommand(command) {
+  return () => {
+    expect(() => {
+      const d = new Directo();
+      d.addCommand(command);
+    }).toThrow();
+  }
+}
+
+function goodCommand(command) {
+  return () => {
+    const d = new Directo();
+    d.addCommand(command);
+  }
+}
+
+test('rejects missing verb',     badCommand({ }));
+test('rejects malformed verb',   badCommand({ verb: 1 }));
+test('rejects missing func',     badCommand({ verb: 'test' }));
+test('rejects malformed func',   badCommand({ verb: [ 'test' ], func: 1 }));
+test('rejects malformed accept', badCommand({ verb: 'test', accept: 2, func() { } }));
+test('rejects uknown accept',    badCommand({ verb: 'test', accept: [ 'uuuh, what' ], func() { } }));
+test('accepts empty accept',     badCommand({ verb: ['go', 'walk' ], accept: [], func() { } }));
+
+test('accepts single verb',      goodCommand({ verb: 'test', func() { } }));
+test('accepts many verbs',       goodCommand({ verb: ['go', 'walk' ], func() { } }));
+test('accepts valid format',     goodCommand({ verb: ['go', 'walk' ], accept: [ Format.V ], func() { } }));
+test('accepts multiple formats', goodCommand({ verb: ['go', 'walk' ], accept: [ Format.V, Format.VO ], func() { } }));
+
+test('rejets commands with duplicate verbs', () => {
+  const d = new Directo();
+  d.addCommand({ verb: ['test', 'go'], func() { } });
+
+  expect(() => {
+    d.addCommand({ verb: 'go', func() { } });
+  }).toThrow();
+});
+
+test('basic processTexting works', async () => {
+  const d = new Directo();
+  let calledEat = false;
+  let calledWalk = false;
+
+  d.addCommand({
+    verb: 'eat',
+    async func() { calledEat = true; }
+  });
+
+  d.addCommand({
+    verb: 'walk',
+    async func() { calledWalk = true; }
+  });
+
+  await d.processText('eat something');
+  expect(calledEat).toBe(true);
+  expect(calledWalk).toBe(false);
+});
+
+test('respects accept', async () => {
```

Comment:
This is `jest`? I'm used to a `describe('thing')` and `it('should do a thing')` syntax for Jest.

Reply:
🤷‍♂️ this is what came up for the first Google lol. I actually prefer the `it` syntax from other frameworks, so I can switch it if jest supports it